src/main/java/com/github/manolo8/darkbot/gui/titlebar/TrayButton.java

```diff
@@ -1,30 +1,107 @@
 package com.github.manolo8.darkbot.gui.titlebar;
 
+import com.github.manolo8.darkbot.Main;
 import com.github.manolo8.darkbot.core.manager.HeroManager;
 import com.github.manolo8.darkbot.gui.MainGui;
-import com.github.manolo8.darkbot.gui.utils.Popups;
 import com.github.manolo8.darkbot.gui.utils.UIUtils;
 import com.github.manolo8.darkbot.utils.I18n;
 
 import javax.swing.*;
 import java.awt.*;
 import java.awt.event.ActionEvent;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
 
 public class TrayButton extends TitleBarButton<JFrame> {
 
     private final TrayIcon icon;
+    private final JPopupMenu popupMenu;
+    private final JDialog dialog;
+
     private boolean shownMsg;
+    private Main main;
 
-    TrayButton(JFrame frame) {
+    TrayButton(Main main, JFrame frame) {
         super(UIUtils.getIcon("tray"), frame);
         setToolTipText(I18n.get("gui.tray_button"));
         setVisible(SystemTray.isSupported());
-        icon = new TrayIcon(MainGui.ICON, "DarkBot");
+
+        this.main = main;
+        popupMenu = createPopup();
+        dialog = createDialog();
+        icon = createTrayIcon();
+    }
+
+    private TrayIcon createTrayIcon() {
+        TrayIcon icon = new TrayIcon(MainGui.ICON, "DarkBot");
         icon.setImageAutoSize(true);
+
         icon.addActionListener(l -> {
             SystemTray.getSystemTray().remove(icon);
             frame.setVisible(true);
+            super.setVisible(true);
+        });
+
+        icon.addMouseListener(new MouseAdapter() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                if (SwingUtilities.isRightMouseButton(e)) {
+                    popupMenu.setLocation(e.getX(), e.getY() - popupMenu.getPreferredSize().height);
+                    popupMenu.setInvoker(popupMenu);
+                    dialog.setLocation(e.getX(), e.getY());
+
+                    dialog.setVisible(true);
+                    popupMenu.setVisible(true);
+                }
+            }
+        });
+        return icon;
+    }
+
+    // Using JPopupMenu because Look And Feel doesn't apply to PopupMenu
+    private JPopupMenu createPopup() {
+        JPopupMenu popup = new JPopupMenu("DarkBot");
+
+        JMenuItem title = new JMenuItem("DarkBot", UIUtils.getIcon("icon"));
+        JMenuItem quit = new JMenuItem("Quit DarkBot");
```

Comment:
Needs to be translated and come from strings file, something like:
```suggestion
        JMenuItem quit = new JMenuItem(I18n.get("gui.tray_menu.quit");
```